### DISCUSSION TITLE
로그인 기능을 구현해본 경험이 있나요?

### DISCUSSION THREAD

**Discussion by WooGie911** [wanted-9th-3team/frontend-study#18] [프론트 기술 질문 공간]
로그인 기능을 구현해본 경험이 있다면 경험과 사용한 기술을 설명해주세요.

**Comment by michoball** [comment]
기본적인 로그인 방식과 소셜 로그인 방식으로 구현을 했었습니다. 
기본적인 로그인을 구현할 때 validation은 복잡한 룰을 가지지 않는 선에서 구현을 했었습니다. 
( 이메일: @포함, 비밀번호: 8자이상 ) 그러나 따로 요구사항이 있으면 정규표현식을 사용하여 validation check를 했었고, 최근에 react-hook-form 라이브러리을 알게되어 앞으로 구현을 한다면 사용해볼 의향이 있습니다.  -> 안써본 라이브러리 언급은 하지 말아야할까요 ?

로그인 한 유저의 상태관리는 redux-toolkit의 createAsyncThunk를 사용하거나 react-query를 사용했었습니다.

사용자 인증 방식은 전통적인 쿠키 세션 방식을 사용하기보다 JWT를 이용, google 네이버 카카오로 Oauth 로그인을 구현한 경험이 있습니다.

**Comment by WooGie911** [comment]
저는 이메일 인증 형식의 로그인 방식과 OpenAPI를 이용한 소셜 로그인 방식을 구현해 본 경험이 있습니다.

실제 프로젝트에서 구현한 이메일 인증 형식은 백엔드에서 라이브러리를 사용해 백엔드 api에서 보내준 인증 값과 비교하여 로그인 해주는 방식이었으며 OpenAPI를 이용한 소셜 로그인 방식은 카카오 OpenAPI를 사용하여 Oauth 로그인을 구현한 경험이었습니다.

nodemailer를 사용해 이메일 인증 방식은 프로젝트 경험은 없고 개인적으로 연습해본 경험이 있습니다.

**Comment by BaekYuri** [comment]
로그인 시, axios를 이용해 http 요청을 하고, 성공 시 백엔드에서 JWT 토큰을 받아 로컬 스토리지에 저장하는 방식으로 구현하였습니다.
다른 api 요청을 할 때, 헤더에 발급받았던 JWT 토큰을 같이 추가하여 요청을 하였습니다.

**Comment by skyhanull** [comment]
스택오버플로우라는 사이트를 클론코딩 시 구현을 해본 경험이 있습니다.
jwt를 이용한 인증방식을 사용했는데 이 때 상태관리 툴을 recoil로 사용하여 들어온 로그인 정보를 Persist를 이용해 로컬스토리지에 저장했습니다 그리고 cookie로도 구현을 했었는데 cookie는 유니버셜 쿠키를 통해 구현했습니다
유효성 검사는 백엔드에서 한번 프론트엔드에서 한번 구현을 했습니다 프론트에서는 정규식을 이용하여 맞지 않을 시에 경고문을 보여주었습니다.
